hail/src/main/scala/is/hail/expr/ir/MatrixWriter.scala

```diff
@@ -210,30 +218,58 @@ case class SplitPartitionNativeWriter(
     r.union(streamType.required)
   }
 
-  if (stageLocally)
-    throw new LowererUnsupportedOperation("stageLocally option not yet implemented")
-  def ifIndexed[T >: Null](obj: => T): T = if (hasIndex) obj else null
-
-  def consumeStream(
-    ctx: ExecuteContext,
-    cb: EmitCodeBuilder,
-    stream: StreamProducer,
-    context: EmitCode,
-    region: Value[Region]): IEmitCode = {
+  def consumeStream(ctx: ExecuteContext,
+                    cb: EmitCodeBuilder,
+                    stream: StreamProducer,
+                    context: EmitCode,
+                    region: Value[Region]
+                   ): IEmitCode = {
     val iAnnotationType = PCanonicalStruct(required = true, "entries_offset" -> PInt64())
     val mb = cb.emb
 
-    val indexWriter = ifIndexed { StagedIndexWriter.withDefaults(index.get._2, mb.ecb, annotationType = iAnnotationType,
-      branchingFactor = Option(mb.ctx.getFlag("index_branching_factor")).map(_.toInt).getOrElse(4096)) }
+    val writeIndexInfo = index.map { case (name, ktype) =>
```

Comment:
I'm allergic to `null` and have been bitten by them too many times. Keeping things within the `Option` allows us to avoid things like null refs and `ifIndexed`